### PR TITLE
[MLIR][TORCH] Add support for 1-d group convolution

### PR DESCRIPTION
**Pull request by vivekkhandelwal1**
This commit adds the support for the 1-d depthwise convolution as a special case of 1-d group convolution.

Signed-Off By: Vivek Khandelwal <vivekkhandelwal1424@gmail.com>

### CODE REVIEW

File: lib/Conversion/TorchToLinalg/Linear.cpp [hunk 1]
```diff
@@ -1236,21 +1232,45 @@ class ConvertAtenConvolutionOp : public OpConversionPattern<AtenConvolutionOp> {
...
+                         loc, outputTensor.getType(),
+                         ValueRange{paddedInput, collapsedWeight}, outputTensor,
+                         stridesAttr, dilationAttr)
+                     .getResult(0);
+          break;
+        default:
+          return rewriter.notifyMatchFailure(
+              op, "unimplemented: only 1D and 2D depthwise convolution "
+                  "supported for special case of group convolution");
+        };
```
Comment (kumardeepakamd) [inline lib/Conversion/TorchToLinalg/Linear.cpp:1267]: return success()" here too?

**Review by kumardeepakamd — APPROVED** [review]
It is covering only the case that can be depthwise 1d conv, and using preexisting function. Good to go.

File: lib/Conversion/TorchToLinalg/Linear.cpp [hunk 2]
```diff
@@ -1297,6 +1317,10 @@ class ConvertAtenConvolutionOp : public OpConversionPattern<AtenConvolutionOp> {
       return success();
     }
 
+    if (numSpatialDims != 2)
+      return rewriter.notifyMatchFailure(
+          op, "unimplemented: only 2D grouped convolution supported");
+
```
[the remark this answers is not in the document]
Comment (vivekkhandelwal1) [reply lib/Conversion/TorchToLinalg/Linear.cpp:1323]: Yes

File: lib/Conversion/TorchToLinalg/Linear.cpp (hunk 1 above)
Comment (vivekkhandelwal1) [reply lib/Conversion/TorchToLinalg/Linear.cpp:1267]: No, there's a common return for both the if and else path.

**Pull request merged**